seckill-command-service/src/main/java/io/servicecomb/poc/demo/seckill/SecKillPromotionBootstrap.java

```diff
@@ -34,63 +33,62 @@
 
   private static final Logger logger = LoggerFactory.getLogger(SecKillPromotionBootstrap.class);
 
-  private final PromotionRepository promotionRepository;
-  private final PromotionEventRepository<T> eventRepository;
+  private final SpringPromotionRepository promotionRepository;
   private final Map<String, SecKillCommandService<T>> commandServices;
-  private final List<SecKillPersistentRunner<T>> persistentRunners;
+  private final List<SecKillEventPersistentRunner<T>> persistentRunners;
+  private final SecKillEventPersistent eventPersistent;
   private final SecKillRecoveryService<T> recoveryService;
 
-  private final ScheduledExecutorService executorService = Executors.newScheduledThreadPool(1);
+  private final Map<String, PromotionEntity> waitingPromotions = new HashMap<>();
   private int loadedPromotionId = 0;
-  private final Map<String, Promotion> waitingPromotions = new HashMap<>();
 
   public SecKillPromotionBootstrap(
-      PromotionRepository promotionRepository,
-      PromotionEventRepository<T> eventRepository,
+      SpringPromotionRepository promotionRepository,
       Map<String, SecKillCommandService<T>> commandServices,
-      List<SecKillPersistentRunner<T>> persistentRunners,
+      List<SecKillEventPersistentRunner<T>> persistentRunners,
+      SecKillEventPersistent eventPersistent,
       SecKillRecoveryService<T> recoveryService) {
     this.promotionRepository = promotionRepository;
-    this.eventRepository = eventRepository;
     this.commandServices = commandServices;
     this.persistentRunners = persistentRunners;
+    this.eventPersistent = eventPersistent;
     this.recoveryService = recoveryService;
   }
 
   public void run() {
     final Runnable executor = () -> {
-      Iterable<Promotion> promotions = promotionRepository.findByIdGreaterThan(loadedPromotionId);
-      for (Promotion promotion : promotions) {
+      Iterable<PromotionEntity> promotions = promotionRepository.findByIdGreaterThan(loadedPromotionId);
+      for (PromotionEntity promotion : promotions) {
```

Comment:
is it better to filter out promotions with finished event here, to avoid create their corresponding persistent runner?

Reply:
Done